tests/componenttests/server/tests/sessionServerStateChange/SessionServerStateChangeTest.cpp

```diff
@@ -57,3 +106,31 @@ TEST_F(SessionServerStateChangeTest, ShouldConfigureInActiveState)
     willConfigureSocket();
     configureSutInActiveState();
 }
+
+TEST_F(SessionServerStateChangeTest, ShouldChangeFromInactiveToActive)
+{
+    willConfigureSocket();
+    configureSutInInactiveState();
+    setStateActive();
+}
+
```

Comment:
Personally I would combine ShouldConfigureInInactiveState & ShouldChangeFromInactiveToActive and  ShouldConfigureInActiveState &  ShouldChangeFromActiveToInactive, as they are dependant on each other and component tests should be more 'real' world scenarios.

So one test which configures in inactive state and transitions to active state. The other test which configures in the active state and transitions to the inactive state.